### ISSUE TITLE
Add a demo doll to the index page

### ISSUE THREAD

**Issue by emhoracek** [emhoracek/smooch#26]
The index page for the site is a bit bare right now: 

<img width="1003" alt="screen shot 2017-09-30 at 7 38 13 pm" src="https://user-images.githubusercontent.com/5353499/31050388-f4212e84-a616-11e7-900e-c28e29ac37df.png">

Let's add a "demo doll" to the index page, then encourage folks to create an account to play with other sets. See this beautiful mockup:

![smooch](https://user-images.githubusercontent.com/5353499/31050529-e3605580-a61a-11e7-88c2-8805af7e8f34.png)

Please be creative in your interpretation of this mockup. ^_^

The index page can be found at `/haskell/templates/index.tpl`.

To get the code for a demo doll, you can copy code from demos in the `docs` folder. The demos are at `/docs/example-sets/`. I would recommend "yura" because it's a small and simple doll. Copy the folder to `/haskell/static/`, then copy the HTML from that folder into the `index.tpl` template.

It would be even better if you could find an artist that gives explicit permission for their work to be used in this way. I haven't been able to get a hold of any artists so far. -_-

**Comment by mahmedov** [comment]
I'll try to do this one.

**Comment by emhoracek** [comment]
Awesome! Just ask if you run into any problems!

**Comment by mahmedov** [comment]
Hey, I just read the issue and I just want to say that im not the best artist in the world so thats why I removed my comment, sorry.

**Comment by mahmedov** [comment]
I can still work on it using the dolls from smooch but if anyone else who is more talented can do it, then can go ahead.

**Comment by emhoracek** [comment]
@mahmedov Oh, I meant finding an artist like, on [Big KiSS Page](http://otakuworld.com/kiss/) or elsewhere on the interwebs -- I don't mean that the person who claims the issue needs to be an artist. Just putting one of the demo dolls on the index page would be a big improvement!

**Comment by mahmedov** [comment]
Got it! Sorry for misunderstanding.

**Comment by J-Wicks** [comment]
@mahmedov Are you still working on this? If not, I'd like to take a stab at it.

**Comment by mahmedov** [comment]
@J-Wicks You can go ahead and work on it, I am still unable to run it on my machine.

**Comment by emhoracek** [comment]
@mahmedov sorry to hear that! 

@J-Wicks Issue is yours -- let me know if you run into similar problems.

**Comment by J-Wicks** [comment]
@emhoracek Good Morning! When I run the stack build script I'm getting the following error:   /private/var/folders/n0/t0rhfj5n31j673krh_s3bjj00000gn/T/stack77530/postgresql-libpq-0.9.3.1/LibPQ.hsc:214:10: fatal error: 'libpq-fe.h' file not found
    #include <libpq-fe.h>

I'm troubleshooting this right now, but have you encountered it before? Any insights?

**Comment by emhoracek** [comment]
@J-Wicks I *haven't* seen that error before, but I wonder if a step is missing related to the instructions -- do you have Postogresql installed? If so, which version?

**Comment by J-Wicks** [comment]
I'm running 9.6 right now. I've gotten around the above error but I've run into anther that I believe has to do with postgres. I'll let you know if I can't make any progress by this afternoon.

**Comment by J-Wicks** [comment]
I'm now getting the error "Incompatible library version: LibPQ_hsc_make requires version 8.0.0 or later, but libiconv.2.dylib provides version 7.0.0"

I've tried adding extra-include-dirs and extra-lib-dirs to point to usr/local/opt/libiconv but that does not fix the issue. Unfortunately I'm now stuck on this and can't seem to make any headway. Thoughts?

**Comment by emhoracek** [comment]
@J-Wicks I found some ancient google group threads and github issues but no ideas so far. Will investigate further and get back to you!

**Comment by emhoracek** [comment]
@J-Wicks I'm not sure how to resolve this. -_- 

[This post](https://groups.google.com/d/msg/yesodweb/mh7L_Tlmy7k/1MNHgWvzvjsJ) suggests deleting postgres and installing [Postgres.app](https://postgresapp.com/), which seems a bit extreme to me.  Since I don't know why it worked for that person, I can't guarantee that it'll work for you, especially since the post is two years old.

The other thing I found is [this Github issue](https://github.com/lpsmith/postgresql-libpq/issues/9#issuecomment-205040648). It sounds like you already tried that solution? If it didn't work, would you mind making a comment about your situation in that thread?

**Comment by emhoracek** [comment]
@J-Wicks thanks for trying! I will also ping you if I learn anything or if I (or another contributor) gets Vagrant set up or something.

**Comment by aklap** [comment]
@emhoracek I find the template file in: 

`smooch/app/templates/index.tpl` vs. `/haskell/templates/index.tpl`

I'd like to work on this.

**Comment by emhoracek** [comment]
Yep it's moved since I made this issue! I'll update the original post tomorrow.

I'm so excited 😆

The trick with this project is getting it up and running -- so definitely let me know if you have any trouble with it!

**Comment by aklap** [comment]
Setting up the dev environment and running an app is always the hardest part I find, no matter the project 🙂  I'm learning a lot, no worries!

**Comment by aklap** [comment]
Everything is up and running for me 🎉  One question @emhoracek , is there hot reloading with this server? I just quickly googled and it looks like the answer is no?

**Comment by emhoracek** [comment]
@aklap hooray!! 

I usually just manually reload bc I haven't done much frontend work yet. There is a way to get hotreloading working, using `nodemon` (from npm). It would be something like (copying from another of my projects):

`stack build && nodemon --ext tpl --watch templates --watch src --exec "stack exec" app`

But I haven't tried it yet!

**Comment by aklap** [comment]
This worked for me: 

`stack build && nodemon --ext tpl --watch templates --watch src --exec "stack exec" smooch`

**Comment by aklap** [comment]
Sorry for the wait on this one, does this look like what you were thinking of @emhoracek? I'm open to suggestions, since I'm admittedly not a designer. 

![Home page layout](https://i.imgur.com/28yhCh8.png)

**Comment by emhoracek** [comment]
Hi @aklap! I've started working on this project again and wanted to clean up the PRs. I've made a new PR based on this one that cleans up a couple TODOs and merges in GHC updates and I'm going to go ahead and merge it. Thanks so much for your work on this and sorry for not finishing it sooner!
